lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_git_helper.rb

```diff
@@ -36,13 +36,25 @@ def self.check_branch_exists(branch_name)
         !Action.sh("git branch --list #{branch_name}").empty?
       end
 
+      # Update the RELEASE-NOTES.txt to add a new entry for the new version, then commit the changes.
+      #
+      # @env [String] PROJECT_ROOT_FOLDER The env variable containing the project's root folder. Uses current folder if nil.
+      #
+      # @param [String] new_version The new version number to add an entry for
+      # 
       def self.update_release_notes(new_version)
-        Action.sh("cp #{ENV["PROJECT_ROOT_FOLDER"]}RELEASE-NOTES.txt #{ENV["PROJECT_ROOT_FOLDER"]}RELEASE-NOTES.bak")
-        Action.sh("echo \"#{new_version}\n-----\n \" > #{ENV["PROJECT_ROOT_FOLDER"]}RELEASE-NOTES.txt")
-        Action.sh("cat #{ENV["PROJECT_ROOT_FOLDER"]}RELEASE-NOTES.bak >> #{ENV["PROJECT_ROOT_FOLDER"]}RELEASE-NOTES.txt")
-        Action.sh("git add #{ENV["PROJECT_ROOT_FOLDER"]}RELEASE-NOTES.txt")
-        Action.sh("git diff-index --quiet HEAD || git commit -m \"Update release notes\"")
-        Action.sh("git push origin HEAD")
+        path = File.join(ENV["PROJECT_ROOT_FOLDER"] || '.', 'RELEASE-NOTES.txt')
+        lines = File.readlines(path)
+        # Find the index of the first non-empty line that is also NOT a comment. That way we keep commment headers as the very top of the file
+        line_idx = lines.find_index { |l| !l.start_with?('***') && !l.start_with?('//') && !l.chomp.empty? }
+        # Put back the header, then the new entry, then the rest (note: '...' excludes the higher bound of the range, unlike '..')
+        new_lines = lines[0...line_idx] + ["#{new_version}\n", "-----\n", "\n", "\n"] + lines[line_idx..]
+        File.write(path, new_lines.join)
```

Comment:
Nice that, on top of the new feature, you replaced `Action.sh` with pure Ruby 👍

Reply:
Yeah the next plan is to extract the remaining `Action.sh` which do the commit + push stuff, so that git-related code is kept in `*_git_helper.rb` but non-git related code like above is moved in a better place/file. But that's something that The Great Refactor™ (paaHJt-1JR-p2) will take care of in a separate PR.